### DISCUSSION TITLE
For novelupdates.com why do I get initial chapter URLs instead of the redirected URLs?

### DISCUSSION THREAD

**Discussion by Mepper65** [maoserr/epublifier#387] [Q&A] [answered]
I'm trying to download from https://www.novelupdates.com/series/became-a-medieval-fantasy-wizard/

maoserr captured the following image:

![EpubEx01](https://github.com/user-attachments/assets/70194487-46a9-4466-8dfe-e6b46d56dc8f)

When I run the latest version of Epublifier on the same page (both Firefox and Chrome) I get:

![EpubEx02](https://github.com/user-attachments/assets/844fe78f-b7d3-4094-b2c2-002fbeb2398b)

The differences:

1. maoserr # column values start at 0 and increase, mine start at 160 and decrease.  Probably immaterial, but a difference.

2. maoserr title column shows a single c1 entry, mine shows 2x c1 entries.  Novelupdates actually has 2 chapter 1s, the first by a different translator.  Maybe the first c1 was manually deleted from the list?

3. maoserr url column shows links from raeitranslations.com, mine shows links from www.novelupdates.com.

I know that for the second chapter 1
https://www.novelupdates.com/extnu/8121498/
redirects to
https://raeitranslations.com/became-a-medieval-fantasy-wizard/1
so it appears that I am picking up the initial URLs from novelupdates while maoserr is picking up the final redirected URLs.

I suspect whatever is causing difference #3 is what is preventing Epublifier from working for me.

Any idea why this is happening?

**Comment by maoserr** [comment, marked as answer]
you don't start from novel updates, you start from chapter 1 of the novel

**Comment by Mepper65** [comment]
I'm missing something very basic...

If I go to https://raeitranslations.com/became-a-medieval-fantasy-wizard/1 (the first chapter), run Epublifier, switch it to parse as an app instead of pages, then go to the "Add page parser" and set the next and title selectors.  Finally I click the + button to add the page plus the following x pages to the chapter list.  So far so good.  I now have a result that looks just like the image you had.

I then try to parse a chapter.  It doesn't matter which one.  I get the "Cannot read properties of null (reading 'title')" error on Chrome or "out is null" on Firefox.
Log from Chrome:
Chapter 0
Parsing chapter content: 0
Error: 1 errors
Cannot read properties of null (reading 'title')

On the "Chapter Parser" tab I have "[main] Readability" selected and no parser options specified.
If I'm interpreting it correctly it is saying the app is trying to access the 'title' attribute of an object that hasn't been initialized.

Alternatively, instead of going to https://raeitranslations.com/became-a-medieval-fantasy-wizard/1 as above, I can go to the main page for the book at https://raeitranslations.com/became-a-medieval-fantasy-wizard, load the chapters, then start Epublifiier, go to the "Links Parser" tab and click the "(Re)Parse Links" or "Add Links" button.  This gets me the complete list of chapter URLs without loading and parsing all the pages looking for "next" links.  Of course as soon as I try parsing a chapter I get the same error as in the first method.

I realize that not all book sites will have a top page that lists all chapter URLs in a way that Epublifier can find & use them, but in this case isn't it a better approach?

Regardless, I can't successfully parse a chapter and save it into an epub.

I tried going into the Parser Definition and adding writes to the console in each method, but none of the writes showed up in the log.  I also tried increasing and decreasing the index in the title selector just in case the page contents were slightly different from what I saw in my browsers.

Any thoughts on what I should try next?

**Reply by maoserr** [reply]
Chapters are already parsed once you click the + button. You just set the max chapter and click the + button to parse that amount of chapters.

Then you can generate epub. 

Parse button is not needed for this scenario.

**Comment by Mepper65** [comment]
Thanks.  I should have noticed the check marks and figured it out.  Everything is working now.
